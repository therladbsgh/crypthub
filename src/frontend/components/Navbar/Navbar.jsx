import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { UserBackend } from 'endpoints';
import { Menu, Dropdown, Button, Icon } from 'semantic-ui-react';
import { NavbarStyle as styles } from 'styles';

class Navbar extends Component {
    constructor(props) {
        super(props);

        this.logout = this.logout.bind(this);
    }

    logout() {
        const { history } = this.props;

        UserBackend.logout()
		.then(res => {
            console.log('success! ', res);
            history.push('/');
		}, ({ err }) => {
            console.log('error! ', err);
            alert(`Logout error: ${err}`);
        });
    }

  	render() {
        const { username } = this.props;
        
		return (
            <Menu id={styles.navbar} inverted>
                <Menu.Item name='home' as={Link} to={username ? '/games' : '/'}>
                    <Icon name='home' />CryptHub
                </Menu.Item>

                <Menu.Menu position='right'>
                    <Menu.Item as={Link} to='/rankings'>
                        <Icon name='trophy' />Global Rankings
                    </Menu.Item>
                    <Menu.Item as={Link} to='/docs'>
                        <Icon name='file code outline' />API Documentation
                    </Menu.Item>
                    {username ? 
                    <Dropdown item text={username}>
                        <Dropdown.Menu>
                            <Dropdown.Item as={Link} to='/games'><Icon name='game' />Games</Dropdown.Item>
                            <Dropdown.Item><Icon name='setting' />Settings</Dropdown.Item>
                            <Dropdown.Item onClick={this.logout}><Icon name='sign out' />Logout</Dropdown.Item>
                        </Dropdown.Menu>
                    </Dropdown>
                    :
                    [<Menu.Item key='signup' as={Link} to='/signup'>
                        <Icon name='signup' />Signup
                    </Menu.Item>,
                    <Menu.Item key='login' as={Link} to='/login'>
                        <Icon name='sign in' />Login
                    </Menu.Item>]}
                </Menu.Menu>
            </Menu>
		);
  	}
}

export default withRouter(Navbar);
