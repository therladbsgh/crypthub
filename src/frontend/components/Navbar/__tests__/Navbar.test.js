import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Navbar';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
  	expect(tree).toMatchSnapshot();
});