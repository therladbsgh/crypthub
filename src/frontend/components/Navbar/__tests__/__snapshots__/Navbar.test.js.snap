// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui inverted menu"
  id={undefined}
>
  <a
    className="item"
    href="/"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="home icon"
    />
    CryptHub
  </a>
  <div
    className="right menu"
  >
    <a
      className="item"
      href="/rankings"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="trophy icon"
      />
      Global Rankings
    </a>
    <a
      className="item"
      href="/docs"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="file code outline icon"
      />
      API Documentation
    </a>
    <a
      className="item"
      href="/signup"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="signup icon"
      />
      Signup
    </a>
    <a
      className="item"
      href="/login"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="sign in icon"
      />
      Login
    </a>
  </div>
</div>
`;
